src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,65 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
-    public void getStatistic(String fromFileName, String toFileName) {
 
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String DELIMITER = ",";
+    private static final String LINE_SEPARATOR = System.lineSeparator();
+
+    public String getStatistic(String fromFileName, String toFileName) {
+        int[] totals = readTotals(fromFileName);
+        String report = buildReport(totals[0],totals[1]);
+        writeToFile(toFileName,report);
+        return report;
+    }
+
+    private int[] readTotals(String fromFileName) {
+        int supplyTotal = 0;
+        int buyTotal = 0;
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String value = reader.readLine();
+            while (value != null) {
+                String[] part = value.split(DELIMITER);
+                if (part.length != 2) {
+                    throw new RuntimeException("Malformed line: " + value);
+                }
+                if (part[0].equals(SUPPLY)) {
+                    supplyTotal += Integer.parseInt(part[1]);
+                }
+                if (part[0].equals(BUY)) {
+                    buyTotal += Integer.parseInt(part[1]);
+                }
```

Lines with unexpected operation types are silently ignored (only `supply` and `buy` are accumulated, others fall through). This violates: "Lines with unexpected operation types should be treated as invalid input and should cause an exception." Instead of two separate `if` checks, after trimming use an `if (op.equals(SUPPLY)) { ... } else if (op.equals(BUY)) { ... } else { throw new RuntimeException("Unexpected operation '" + op + "' in file " + fromFileName + ": " + value); }` so unknown operations fail fast with a clear message.